README.md

```diff
@@ -1,25 +1,47 @@
 # Rancher Desktop Agent
 
 The Rancher Desktop guest agent runs in Rancher Desktop VMs providing helper
-services.  It currently has the following functionality:
+services. It currently has the following functionality:
+
+## Port forwarding with privileged helper:
+
+When the Rancher Desktop Privileged Service is enabled on the host Windows machine via an admin installation of Rancher Desktop, the guest agent watches for port binding events from corresponding backend's API and emits them to the host via a virtual tunnel.
+
+```mermaid
+flowchart  LR;
+ subgraph Host["HOST"]
+ rd{"Rancher Desktop"}
+ ps(("Privileged Service"))
+ rd <-.-> ps
+ end
+ subgraph VM["WSL VM"]
+ rdagent(("Guest Agent"))
+ api(("moby/containerd \n events API"))
+ rdagent <-.->  api
+ end
+ ps  <---> |Vtunnel| rdagent
+```
+
+### moby port forwarding (WSL)
+
+Rancher Desktop Guest Agent subscribes to docker event API to monitor the newly created published ports. It will then forwards the newly published ports over a `AF_VSOCK` tunnel (Rancher Desktop's `vtunnel`) to [Rancher Desktop Privileged Service](https://github.com/rancher-sandbox/rancher-desktop/tree/main/src/go/privileged-service) that runs on the host machine.
```

Comment:
The `docker event API` bit should link to [their documentation](https://docs.docker.com/engine/api/v1.41/#tag/System/operation/SystemEvents).

I don't see an equivalent for containerd…

Reply:
Please consider wrapping the long lines; they have no effect on the output but makes looking at diffs easier.